app/models/order.rb

```diff
@@ -23,4 +23,16 @@ def has_item
   def purchaser_name
     user.full_name
   end
+  
+  def purchaser_email
+    user.email
+  end
+  
+  def total
+    items.map{ |item| item.price }.inject(:+)
```

Comment:
Space missing to the left of {.<br>Pass `&:price` as an argument to `map` instead of a block.